File: XOR Queries of a Subarray.go
func xorQueries(arr []int, queries [][]int) []int {
    xorr := 0
    dp := []int{0}
    n := len(arr)
    for i := 0; i < n; i++ {
        xorr ^= arr[i]
        dp = append(dp, xorr)
    }

    ans := make([]int, 0)
    for _, query := range queries {
        ans = append(ans, dp[query[1]+1] ^ dp[query[0]])
    }
    return ans
}

File: XOR Queries of a Subarray.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        xorr = 0
        dp = [0]
        n = len(arr)
        for i in range(n):
            xorr ^= arr[i]
            dp.append(xorr)

        ans = []
        for (left, right) in queries:
            ans.append(dp[right + 1] ^ dp[left])
        return ans
